### PR TITLE
Migration to hyper 1.0

### LINKED ISSUE #182: Migration from hyper 1.0-rc4 to hyper 1.0
Hyper 1.0 is now stable. Major frameworks are migrating to it.
For the moment v0.14 will still remain the default but the hyper-1 feature deps need to be updated to stable.
Also because http is now 1.0.0, [extensions are `cloneable`](https://github.com/hyperium/http/blob/master/CHANGELOG.md#100-november-15-2023) so it will be possible to clone `extensions` when using the ws transport

Salvo tracking issue : https://github.com/salvo-rs/salvo/issues/494
Axum release PR : https://github.com/tokio-rs/axum/pull/2354

### LINKED ISSUE #186: Compilation problems when trying to use axum 0.7
Probably a dup of #182.

I just updated to the latest stable axum 0.7 and now I get a compilation error when I try to use the layer with axum (see code snippet below).

**Describe the bug**
Can't compile with socketioxide and axum 0.7.

**To Reproduce**

```
#[tokio::main]
async fn main() {
  ...
    let (layerx, io) = SocketIo::new_layer();

    // Set up the HTTP server.
    let app: Router = Router::new()
        ...
        .merge(blocks_medias::routes::routes())
        .layer(layerx)
        .layer(DefaultBodyLimit::max(1024 * 1024 * 200))
        .with_state(state);

    let listener = tokio::net::TcpListener::bind("0.0.0.0:3000").await.unwrap();
    axum::serve(listener, app).await.unwrap();
}

```

**Compiler error:**

```
error[E0277]: the trait bound `SocketIoService<Route>: tower_service::Service<axum::http::Request<axum::body::Body>>` is not satisfied
   --> src/main.rs:78:16
    |
78  |         .layer(layerx)
    |          ----- ^^^^^^ the trait `tower_service::Service<axum::http::Request<axum::body::Body>>` is not implemented for `SocketIoService<Route>`
    |          |
    |          required by a bound introduced by this call
    |
    = help: the trait `tower_service::Service<http::request::Request<ReqBody>>` is implemented for `SocketIoService<S, A>`
note: required by a bound in `Router::<S>::layer`
   --> /Users/bbense/.cargo/registry/src/index.crates.io-6f17d22bba15001f/axum-0.7.1/src/routing/mod.rs:236:21
    |
233 |     pub fn layer<L>(self, layer: L) -> Router<S>
    |            ----- required by a bound in this associated function
...
236 |         L::Service: Service<Request> + Clone + Send + 'static,
    |                     ^^^^^^^^^^^^^^^^ required by this bound in `Router::<S>::layer`


```

**Versions (please complete the following information):**
 - Socketioxide version: head and 0.7.2
 - axum 0.7

### PR DESCRIPTION
**Pull request by Totodore**
## Migration to hyper 1.0

* The `hyper-v1` feature flag will be removed.
* All hyper dependencies will be upgraded to stable v1. Therefore the current release won't work with `salvo` until it migrates (https://github.com/salvo-rs/salvo/pull/520).
* ~~An adapter for `hyper-legacy` maybe worked so that frameworks that didn't migrated yet such as `warp` can still work. However there is some difficulty with the adaptation between `http` 1.0 types and `http` 0 types.~~

* Fixes #186 
* Fixes #182 

## Remaining tasks : 
- [x] Update doc and make a compatibility table with all http frameworks.

### CODE REVIEW

**Comment by Totodore** [comment]
> Awesome to see some progress! Looking forward to the update.

It should be released today or this weekend.

File: examples/angular-todomvc/src/main.rs [hunk 1]
```diff
@@ -65,11 +63,8 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
                 .layer(layer),
         );
 
-    let server = Server::bind(&"0.0.0.0:8080".parse().unwrap()).serve(app.into_make_service());
-
-    if let Err(e) = server.await {
-        error!("server error: {}", e);
-    }
+    let listener = tokio::net::TcpListener::bind("0.0.0.0:3000").await.unwrap();
```
Comment (fundon) [inline examples/angular-todomvc/src/main.rs:66]: port should be `8080`

Comment (Totodore) [reply examples/angular-todomvc/src/main.rs:66]: Can you detail a bit more ? If it is really the case, that is not the subject of this PR and it should be treated in another.

Comment (fundon) [reply examples/angular-todomvc/src/main.rs:66]: The `socke.io` connects to the `8080` port. 
https://github.com/Totodore/socketioxide/blob/d563248f2b983af3ee3a25cff0850c4f3257d8d1/examples/angular-todomvc/dist/main.js#L297

Comment (fundon) [reply examples/angular-todomvc/src/main.rs:66]: Or use CDN path

**Pull request merged**